include/swift/AST/DiagnosticsSema.def

```diff
@@ -1216,7 +1216,9 @@ WARNING(nscoding_unstable_mangled_name_warn,none,
         "%select{private|fileprivate|nested|generic}0 class %1 has an unstable "
         "name when archiving via 'NSCoding'",
         (unsigned, Type))
-
+NOTE(add_nskeyedarchivelegacy_attr,none,
+     "add the '@NSKeyedArchiveLegacy' attribute to specify the class name "
+     "used for archiving", ())
```

Comment:
Can you change this to make this more discouraged? The recommended fix is to use `@objc` (which deserves its own note).

- `"for new classes, add '@objc' to specify a unique, prefixed Objective-C runtime name"`
- `"for compatibility with existing archives, use '@NSKeyedArchiveLegacy' to record the Swift 3 mangled name` (or "internal name", or just "name")

Also, nitpick: I think our diagnostics usually either say `'@foo'` or `'foo' attribute` but not both.